tests/V8Server/v8-analyzers-test.cpp

```diff
@@ -1271,6 +1271,124 @@ TEST_F(V8AnalyzersTest, test_create) {
+    auto& user =
+        userMap
+            .emplace("", arangodb::auth::User::newUser("", "", arangodb::auth::Source::LDAP))
+            .first->second;
+    user.grantDatabase(vocbase.name(), arangodb::auth::Level::RW);  // for system collections User::collectionAuthLevel(...) returns database auth::Level
+    userManager->setAuthInfo(userMap);  // set user map to avoid loading configuration from system database
+
+    auto result = v8::Function::Cast(*fn_create)
+                      ->CallAsFunction(context, fn_create,
+                                       static_cast<int>(args.size()), args.data());
+    EXPECT_TRUE((!result.IsEmpty()));
+    EXPECT_TRUE((result.ToLocalChecked()->IsObject()));
+    auto* v8Analyzer = TRI_UnwrapClass<arangodb::iresearch::IResearchAnalyzerFeature::AnalyzerPool>(
+        result.ToLocalChecked()->ToObject(TRI_IGETC).FromMaybe(v8::Local<v8::Object>()),
+        WRP_IRESEARCH_ANALYZER_TYPE, TRI_IGETC);
+    EXPECT_TRUE((false == !v8Analyzer));
+    EXPECT_EQ(vocbase.name() + "::testAnalyzer3", v8Analyzer->name());
+    EXPECT_EQ("identity", v8Analyzer->type());
+    EXPECT_EQUAL_SLICES(
+        VPackSlice::emptyObjectSlice(),
+        v8Analyzer->properties());
+    EXPECT_TRUE(v8Analyzer->features().empty());
+    auto analyzer = analyzers->get(vocbase.name() + "::testAnalyzer3");
+    EXPECT_TRUE((false == !analyzer));
+  }
+  // successful creation in system db by :: prefix
+  {
+    TRI_vocbase_t vocbase(TRI_vocbase_type_e::TRI_VOCBASE_TYPE_NORMAL, 1,
+                          arangodb::StaticStrings::SystemDatabase);
+    v8::Isolate::CreateParams isolateParams;
+    ArrayBufferAllocator arrayBufferAllocator;
+    isolateParams.array_buffer_allocator = &arrayBufferAllocator;
+    auto isolate =
+        std::shared_ptr<v8::Isolate>(v8::Isolate::New(isolateParams),
+                                     [](v8::Isolate* p) -> void { p->Dispose(); });
+    ASSERT_TRUE((nullptr != isolate));
+    v8::Isolate::Scope isolateScope(isolate.get());  // otherwise v8::Isolate::Logger() will fail (called from v8::Exception::Error)
+    v8::internal::Isolate::Current()->InitializeLoggingAndCounters();  // otherwise v8::Isolate::Logger() will fail (called from v8::Exception::Error)
+    v8::HandleScope handleScope(isolate.get());  // required for v8::Context::New(...), v8::ObjectTemplate::New(...) and TRI_AddMethodVocbase(...)
+    auto context = v8::Context::New(isolate.get());
+    v8::Context::Scope contextScope(context);  // required for TRI_AddMethodVocbase(...)
+    std::unique_ptr<TRI_v8_global_t> v8g(TRI_CreateV8Globals(isolate.get(), 0));  // create and set inside 'isolate' for use with 'TRI_GET_GLOBALS()'
+    v8g->ArangoErrorTempl.Reset(isolate.get(), v8::ObjectTemplate::New(isolate.get()));  // otherwise v8:-utils::CreateErrorObject(...) will fail
+    v8g->_vocbase = &vocbase;
+    arangodb::iresearch::TRI_InitV8Analyzers(*v8g, isolate.get());
```

Comment:
The test setup is quite large and happens often, can't that easily be extracted?

Reply:
Usually setup for different test suites may be slightly differ, But yes, some common part may be exctracted. But let`s adress it with separate PR as this will affect all test cases for arango 
 search

Reply:
Sure.